runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -236,6 +236,12 @@ void __catalyst__rt__fail_cstr(const char *cstr) { RT_FAIL(cstr); }
 
 void __catalyst__rt__initialize() { CTX = std::make_unique<ExecutionContext>(); }
 
+void __catalyst__rt__initialize_seeded(char *seed)
+{
+    std::string seed_str(seed);
+    CTX = std::make_unique<ExecutionContext>(seed_str);
+}
+
```

Comment:
Is it not possible to add this to the initialize method above instead of creating a new method?

Reply:
Not if we want the unseeded `quantum.init` to still generate a `__catalyst__rt__initialize()` that takes in no arguments. The mlir will have a declaration that takes in no arguments (there are mlir tests that check this), so we cannot have just one `void __init(char *=nullptr)` and set default arguments, since the function types will not match.

Reply:
What I'm thinking is:

*  in the setup function write the MLIR in LLVM dialect that will call / or modify a static variable in the runtime.
* during __catalyst__rt__initialize retrieve this seed and use it to generate the execution context.

What do you think?

Reply:
Or just add a function similar to setup that just returns the seed and call it from __catalyst__rt__initialize. Declare it extern in the runtime. If no seed is provided by the user.

Reply:
Personally I like the approach of using `void __catalyst__rt__initialize(char *)` everywhere and let some protected default value handle the normal use.

Reply:
Sounds fine by me, it does simplify our implementations I think :)